CRM/Financial/BAO/FinancialItem.php

```diff
@@ -295,13 +295,28 @@ public static function checkContactPresent($contactIds, &$error) {
    * @return object CRM_Core_DAO
    */
   public static function getPreviousFinancialItem($entityId, $entityTable = 'civicrm_line_item') {
+    $liabilityAccountType = CRM_Core_OptionGroup::getValue('financial_account_type', 'Liability', 'name');
     $queryParams = array(
       1 => array($entityId, 'Integer'),
       2 => array($entityTable, 'String'),
+      3 => array($liabilityAccountType, 'Integer'),
     );
-    $query = 'SELECT id, description, status_id, financial_account_id 
-      FROM civicrm_financial_item
-      WHERE entity_id = %1 AND entity_table = %2 ORDER BY id DESC LIMIT 1';
+    $query = 'SELECT cfi.id, cfi.description, cfi.amount, cfi.financial_account_id, cfi.status_id
+      FROM `civicrm_financial_item` cfi
+        INNER JOIN civicrm_entity_financial_trxn ceft 
+          ON ceft.entity_id = cfi.id AND ceft.entity_table = "civicrm_financial_item"
+        INNER JOIN civicrm_financial_trxn cft ON cft.id = ceft.financial_trxn_id
+        INNER JOIN civicrm_financial_account cfa ON cfa.id = cfi.financial_account_id
+      WHERE cfi.entity_table = %2 AND cfi.entity_id = %1 AND 
+        CASE 
```

Comment:
is this a complete query , because I don't see any meaning for this case statement .. whenever any case get satisfied it will end up either as ( AND 1 ) or (AND 0) and either way it has no meaning since you are not comparing or use it in anything .